File: LeetCode/cpp/2221_Find_Triangular_Sum_of_an_Array.cpp
class Solution {
public:
	int triangularSum(vector<int>& nums) {
		int answer = 0;

		vector<int> v;

		while (nums.size() > 1) {
			for (int i = 0; i < nums.size() - 1; i++) {
				v.push_back((nums[i] + nums[i + 1]) % 10);
			}

			nums.clear();
			for (int n : v) {
				nums.push_back(n);
			}

			v.clear();
		}

		answer = nums[0];
		return answer;
	}
};

File: LeetCode/java/2221_Find_Triangular_Sum_of_an_Array.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int triangularSum(int[] nums) {
        int answer = 0;

        List<Integer> list = new ArrayList<>();
        List<Integer> v = new ArrayList<>();

        for (int n : nums) {
            list.add(n);
        }

        while (list.size() > 1) {
            for (int i = 0; i < list.size() - 1; i++) {
                v.add((list.get(i) + list.get(i + 1)) % 10);
            }

            list.clear();
            for (int n : v) {
                list.add(n);
            }

            v.clear();
        }

        answer = list.get(0);
        return answer;
    }
}